gap/semigroups/semicons.gi

```diff
@@ -703,3 +705,97 @@ function(arg)
   od;
   return Semigroup(gens, rec(acting := false));
 end);
+
+InstallGlobalFunction(BrandtSemigroup,
+function(arg)
+  local S;
+
+  if Length(arg) = 1 and IsPosInt(arg[1]) then
+    S := BrandtSemigroupCons(IsPartialPermSemigroup, Group(()), arg[1]);
+  elif Length(arg) = 2 and IsOperation(arg[1]) and IsPosInt(arg[2]) then
+    S := BrandtSemigroupCons(arg[1], Group(()), arg[2]);
+  elif Length(arg) = 2 and IsGroup(arg[1]) and IsPosInt(arg[2]) then
+    S := BrandtSemigroupCons(IsPartialPermSemigroup, arg[1], arg[2]);
+  elif Length(arg) = 3 and IsOperation(arg[1]) and IsGroup(arg[2])
+      and IsPosInt(arg[3]) then
+    S := BrandtSemigroupCons(arg[1], arg[2], arg[3]);
+  else
+    ErrorNoReturn("Semigroups: BrandtSemigroup: usage,\n",
+                  "the arguments must be a positive integer or ",
+                  "a filter and a positive integer, or a perm group and ",
+                  "positive integer, or a filter, perm group, and positive ",
+                  "integer,");
+  fi;
+  SetIsZeroSimpleSemigroup(S, true);
+  SetIsBrandtSemigroup(S, true);
+  return S;
+end);
+
+InstallMethod(BrandtSemigroupCons,
+"for IsPartialPermSemigroup and a positive integer",
+[IsPartialPermSemigroup, IsPermGroup, IsPosInt],
+function(filter, G, n)
+  local gens, one, m, i, x;
+
+  gens := [];
+
+  if IsTrivial(G) then
+    if n = 1 then
+      Add(gens, PartialPerm([1], [1]));
+      Add(gens, EmptyPartialPerm());
+    else
+      for i in [2 .. n] do
+        Add(gens, PartialPerm([1], [i]));
+      od;
+    fi;
+  else
+    one := PartialPerm(MovedPoints(G), MovedPoints(G));
+    for x in GeneratorsOfGroup(G) do
+      Add(gens, one * x);
+    od;
+    m := LargestMovedPoint(G);
+    for i in [1 .. n - 1] do
+      Add(gens, PartialPerm(MovedPoints(G), MovedPoints(G) + m * i));
+    od;
+    if n = 1 then
+      Add(gens, EmptyPartialPerm());
+    fi;
+  fi;
+
+  return InverseSemigroup(gens);
+end);
+
+InstallMethod(BrandtSemigroupCons,
+"for IsReesZeroMatrixSemigroup and a positive integer",
+[IsReesZeroMatrixSemigroup, IsGroup, IsPosInt],
+function(filter, G, n)
+  local mat, i;
+  mat := [];
+  for i in [1 .. n] do
+    Add(mat, [1 .. n] * 0);
```

Comment:
Now that https://github.com/gap-system/gap/pull/1786 has been merged, I think you should use
`ListWithIdenticalEntries(n, 0)` here instead of `[1 .. n] * 0`. It's clearer, and faster:
```
gap> for i in [1 .. 50000] do x := ListWithIdenticalEntries(n, 0); od; time;
78
gap> for i in [1 .. 50000] do x := [1 .. n] * 0; od; time;
347
```
(Typical performance).

Reply:
I didn't know this, I'll use `ListWithIdenticalEntries` instead